pkg/graveler/ref/commit_iterator.go

```diff
@@ -69,20 +72,20 @@ func NewCommitIterator(ctx context.Context, db db.Database, repositoryID gravele
 	}
 }
 
-func (ci *CommitIterator) getCommitRecord(commitID graveler.CommitID) (*graveler.CommitRecord, error) {
-	var rec commitRecord
-	err := ci.db.
-		Get(ci.ctx, &rec, `SELECT id, committer, message, creation_date, parents, meta_range_id, metadata, version, generation
-			FROM graveler_commits
-			WHERE repository_id = $1 AND id = $2`,
-			ci.repositoryID, commitID)
+func (ci *KVCommitIterator) getCommitRecord(commitID graveler.CommitID) (*graveler.CommitRecord, error) {
```

Comment:
If that's the only function that changed, why duplicate all functions? Especially the very delicate `Next()`

Reply:
because the DB implementation going to be deleted I don't think we need to create a generic function

Reply:
Discussed f2f, let's try to pass a staging manager here